codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt

```diff
@@ -172,11 +177,22 @@ class KotlinSymbolProvider(private val model: Model, private val settings: Kotli
         val targetShape =
             model.getShape(shape.target).orElseThrow { CodegenException("Shape not found: ${shape.target}") }
 
-        val targetSymbol = if (nullableIndex.isMemberNullable(shape, NullableIndex.CheckMode.CLIENT_ZERO_VALUE_V1_NO_INPUT)) {
+        var targetSymbol = if (nullableIndex.isMemberNullable(shape, NullableIndex.CheckMode.CLIENT)) {
             toSymbol(targetShape).toBuilder().boxed().build()
         } else {
             toSymbol(targetShape)
         }
+
+        targetSymbol = shape.getTrait<DefaultTrait>()?.let {
+            val builder = targetSymbol.toBuilder()
+            val defaultValue = it.getDefaultValue(targetShape)
+            builder.defaultValue(defaultValue)
+            if (defaultValue != "null") {
+                builder.unboxed()
+            }
+            builder.build()
+        } ?: targetSymbol
```

Comment:
**Suggestion**: The `var` reference here is unnecessary. You're already dealing with builders is several places so I suggest just keeping it as a builder until you've applied all the customizations necessary:

```kotlin
val targetSymbol = toSymbol(targetShape)
    .toBuilder()
    .apply {
        if (nullableIndex.isMemberNullable(shape, NullableIndex.CheckMode.CLIENT)) boxed()

        shape.getTrait<DefaultTrait>()?.getDefaultValue(targetShape)?.let {
            defaultValue(it)
            if (it != "null") unboxed()
        }
    }
    .build()
```